### PR TITLE
Add support for Python 3.13

### PR DESCRIPTION
**Pull request by rominf**
Fixes https://github.com/celery/kombu/issues/2051.

### CODE REVIEW

File: requirements/extras/confluentkafka.txt
Comment (rominf) [inline requirements/extras/confluentkafka.txt:1]: This line is necessary for the installation of requirements. The change should be removed before the merge.

https://github.com/confluentinc/confluent-kafka-python/issues/1768

File: requirements/extras/zstd.txt
Comment (rominf) [inline requirements/extras/zstd.txt:1]: This line is necessary for the installation of requirements. The change should be removed before the merge.

https://github.com/indygreg/python-zstandard/pull/233

**Comment by auvipy** [comment]
I fixed some merge conflicts, lets see how the integration test for python 3.13 works

**Review by Nusnus — CHANGES_REQUESTED** [review]
https://github.com/celery/kombu/pull/2052#discussion_r1667059593

Should be resolved before merge

**Review by Nusnus — CHANGES_REQUESTED** [review]
```console
ERROR t/integration/test_kafka.py::test_KafkaBasicFunctionality::test_connect - NameError: name 'KafkaException' is not defined
```

File: requirements/extras/zstd.txt
Comment (Nusnus) [inline requirements/extras/zstd.txt:4]:
```suggestion
zstandard==0.23.0; python_version < "3.13"

```

File: requirements/extras/confluentkafka.txt
Comment (Nusnus) [inline requirements/extras/confluentkafka.txt:1]:
```suggestion
confluent-kafka>=2.2.0
```

Comment (Nusnus) [reply requirements/extras/confluentkafka.txt:1]: Let’s see if we can remove it already

File: tox.ini
Comment (Nusnus) [inline tox.ini:140]:
```suggestion
commands = python -m mypy --config-file setup.cfg

```

Comment (Nusnus) [reply tox.ini:140]: Added new line at the end of the file

File: tox.ini
Comment (Nusnus) [inline tox.ini:6]:
```suggestion
    {pypy3.9,3.8,3.9,3.10,3.11,3.12,3.13}-linux-integration-redis
    {pypy3.9,3.8,3.9,3.10,3.11,3.12,3.13}-linux-integration-mongodb
```

Comment (Nusnus) [reply tox.ini:6]: I guess these should have Python 3.12 and 3.13 too

**Review by Nusnus — CHANGES_REQUESTED** [review]
Applying some changes, let’s see how the CI reacts

File: requirements/extras/confluentkafka.txt
Comment (Nusnus) [reply requirements/extras/confluentkafka.txt:1]: Didn’t work:
```console
Failed to build confluent-kafka
```
https://github.com/celery/kombu/actions/runs/10981646035/job/30488936527

**Comment by Nusnus** [comment]
@auvipy I’ve converted it back to Draft as the CI is not ready for merge at the moment.

**Comment by Nusnus** [comment]
> @auvipy I’ve converted it back to Draft as the CI is not ready for merge at the moment.

I am also having issues with pytest-celery: https://github.com/celery/pytest-celery/pull/420

Celery itself did pass the Python 3.13 CI though: https://github.com/celery/celery/pull/9309

File: requirements/extras/zstd.txt
Comment (cclauss) [reply requirements/extras/zstd.txt]: https://github.com/indygreg/python-zstandard/issues/210#issuecomment-2336604856

**Comment by rominf** [comment]
It looks like Google Cloud monitoring library (https://github.com/googleapis/google-cloud-python/tree/main/packages/google-cloud-monitoring) does not support Python 3.13:
```
TypeError: Couldn't build proto file into descriptor pool: duplicate symbol 'google.monitoring.v3.__firstlineno__'
```
See: https://github.com/celery/kombu/actions/runs/11343305734/job/31545591989?pr=2052#step:7:1141

File: .github/workflows/linter.yml
Comment (cclauss) [inline .github/workflows/linter.yml:20]: Does the `lint` job step pass if this change is reverted?

Comment (Nusnus) [reply .github/workflows/linter.yml:20]: Yes

Comment (Nusnus) [reply .github/workflows/linter.yml:20]: But I think tox.ini needs to set 3.12 too if we want to revert it for the lint.

Comment (cclauss) [reply .github/workflows/linter.yml:20]: We can leave in on Py3.13...  We will need to fix it at some point so let's do it now.

**Comment by cclauss** [comment]
Blocked by protocolbuffers/protobuf#18882 or similar.
* protocolbuffers/protobuf#18882

**Review by cclauss — APPROVED** [review]

**Review by Nusnus — APPROVED** [review]
I had a feeling my last rebase would pass due to a recent dep update.

This is finally over, thanks everyone!

**Pull request merged**
